src/Symfony/Component/Messenger/Transport/InMemoryTransport.php

```diff
@@ -36,12 +36,17 @@ class InMemoryTransport implements TransportInterface, ResetInterface
      */
     private $rejected = [];
 
+    /**
+     * @var Envelope[]
+     */
+    private $queue = [];
+
     /**
      * {@inheritdoc}
      */
     public function get(): iterable
     {
-        return $this->sent;
+        return !empty($this->queue) ? [reset($this->queue)] : [];
```

Comment:
why not return the whole queue with `return array_values($this->queue)`? limiting it to the first one seems strange and arbitrary.

Reply:
I have several reasons to do that:
* The [interface](https://github.com/symfony/messenger/blob/master/Transport/Receiver/ReceiverInterface.php#L26) told us to do that way `While this method could return an unlimited number of messages,  the intention is that it returns only one, or a "small number" of messages each time.`
* Other transports like Doctrine or Amqp works that way
* If we return all messages, the `limit` option of the consume command will be useless e.g. `messenger:consume memory --limit=1`

Reply:
That the limit does not work reliably when you return multiple messages is a different problem. It can be solved by just checking if stop() has been called after https://github.com/symfony/symfony/blob/35225902905cc453395685eb90322aa590e707d5/src/Symfony/Component/Messenger/Worker.php#L92 and breaking the loop.
Honestly, I think returning a single message is non-sense when the methods supports returning many. Please revert this. Feel free to create a separate PR with the solution I suggested.

Reply:
Will do, thanks!

Reply:
The PR is https://github.com/symfony/symfony/pull/33298